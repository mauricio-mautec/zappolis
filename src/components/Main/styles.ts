'use client'

import styled from 'styled-components'

export const Wrapper = styled.main`
  background-color: #ffb8b8;
  color: #fff;
  width: 100%;
  height: 100%;
  padding: 3rem;
  text-align: center;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`
