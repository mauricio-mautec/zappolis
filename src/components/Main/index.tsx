import * as S from './styles'

const Main = ({
  title = 'ZappoliS',
  description = 'Commerce Club Community'
}) => (
  <S.Wrapper>
    <S.Logo
      src="/img/zappolis.png"
      alt="Imagem mostra um home de óculos por detrás do escrito ZappoliS"
    />
    <S.Title>{title}</S.Title>
    <S.Description>{description}</S.Description>
    <S.Illustration
      src="/img/zappolis.png"
      alt="Um homem olhando para o computador"
    />
  </S.Wrapper>
)

export default Main
