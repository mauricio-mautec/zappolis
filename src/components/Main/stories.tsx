import { Meta, StoryObj } from '@storybook/react'

import Main from '.'

export default {
  title: 'Main',
  component: Main,
  parameters: {
    layout: 'fullscreen'
  }
} as Meta

export const Default: StoryObj = {
  args: {
    title: 'ZappoliS',
    description: 'Commerce Club Community'
  }
}
