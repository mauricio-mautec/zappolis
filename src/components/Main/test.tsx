import { render, screen } from '@testing-library/react'

import Main from '.'

describe('<Main />', () => {
  it('should render the heading', () => {
    // renderiza o  componente
    const { container } = render(<Main />)
    // busca o elemento e verifica a existencia
    expect(
      screen.getByRole('heading', { name: /ZAPPOLIS/i })
    ).toBeInTheDocument()

    // gerar snapshot
    expect(container.firstChild).toMatchSnapshot()
  })
})
