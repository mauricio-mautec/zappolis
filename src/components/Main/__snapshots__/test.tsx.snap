// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
<main
  class="sc-aYaIB fAXXLU"
>
  <h1>
    ZAPPOLIS MAIN
  </h1>
</main>
`;
